Fixed nilearn dependencies and removed cbar_vmin/vmax redundancies

huzafahyde opened the pull request:
directly imported nilearn functions, and updated the _crop_colorbar function to the newest version. Removed redundant cbar_vmin and cbar_vmax

christianbrodbeck commented on eelbrain/plot/_glassbrain.py:
```diff
@@ -375,6 +415,63 @@ def __init__(
         TimeSlicerEF.__init__(self, 'time', time, display_text=time_text)
         self._show()
 
+    # Copied from nilearn.plotting.img_plotting
```
If you actually changed it (removing `cbar_vmin` if I understood that correctly) you could say "adapted from"

christianbrodbeck reviewed (changes requested):
Great start. I've added some stylistic comments in the code. Let me know if anything is unclear!

I have a request for the functionality: can we make the defaults more intuitive? I've tried using different arguments in the `volume-source-space` example script. For example, using no parameters at all (`plot.GlassBrain(y, title=f"{side} ear tone")`) results in the plot below, using a symmetric color-map but with vmin=0. To be consistent with the rest of the library, the default should probably use `SYMMETRIC_CMAPS` to determine whether `vmin` should be `-vmax` or `0`.

![download](https://github.com/user-attachments/assets/ba47355f-39a1-405c-a737-d7e401d40162)

huzafahyde commented:
Addressed all review comments. For the defaults I have the logic set as such:
If there is no cmap specified by the user, and symmetric_cbar = 'auto' (ie if the user does not specify input parameters), it will set it to True or False depending on the datatype (False if bool, True if not) (lines 384-390). If the user wants to force symmetric_cbar, it will accept that input instead.

christianbrodbeck commented on eelbrain/plot/_glassbrain.py:
```diff
@@ -65,27 +65,126 @@ def get_bounds(shape, affine):
     box = np.dot(affine, box)[:3]
     return list(zip(box.min(axis=-1), box.max(axis=-1)))
 
+# Copied from nilearn.plotting.displays._slicers
```
Could you include two blank lines before function/class definitions (generally we are trying to follow https://peps.python.org/pep-0008/, except for line length, because that often results in somewhat awkward code in my opinion)

christianbrodbeck reviewed (commented):
It's now fixed for the default `cmap`, but when I specify another one I get `vmin=0` – is there a reason not to use `SYMMETRIC_CMAPS` as default?
<img width="842" alt="Screenshot 2025-04-09 at 1 55 28 PM" src="https://github.com/user-attachments/assets/69e91715-bf98-4d16-a806-9a2937bd3eeb" />

christianbrodbeck commented on eelbrain/plot/tests/test_glassbrain.py:
```diff
@@ -20,6 +20,30 @@ def test_glassbrain():
     p.close()
 
+    # vmin, vmax, and symmetric colorbar
+    p = plot.GlassBrain(ndvar_30, colorbar = True, cmap = 'lux-a')
```
```suggestion
    p = plot.GlassBrain(ndvar_30, colorbar=True, cmap='lux-a')
```
Small stylistic change, also below (https://peps.python.org/pep-0008/#other-recommendations)

christianbrodbeck commented:
Looks great now, could you just address the small PEP8 formatting issue? The test error is because the download for a test dataset was failing, I think it's working more reliably again now.
